XenModel/Actions/VMSS/GetVMSSAlertsAction.cs

```diff
@@ -54,16 +54,38 @@ protected override void Run()
             var now = DateTime.Now;
             var messages = Pool.Connection.Cache.Messages;
             var listAlerts = new List<PolicyAlert>();
+ 	    DateTime currentTime = DateTime.Now;
+            DateTime offset = currentTime.Add(new TimeSpan(- _hoursFromNow, 0, 0));
+          
+            /* _hoursFromNow has 3 possible values:
+                1) 0 -> top 10 messages
+                2) 24 -> messages from past 24 Hrs
+                3) 7 * 24 -> messages from lst 7 days */
 
-            /*for VMSS: Populate the alerts from Messages by filtering out the alerts for this schedule
-                This is not required in VMPP as the VMPP record itself has the recentAlerts */
-            foreach (var message in messages)
+            if (_hoursFromNow == 0)
             {
-                if (message.cls == cls.VMSS && message.obj_uuid == VMSS.uuid)
+                int messageCounter = 0;
```

Comment:
The list of messages is not sorted and first 10 items are not necessarly the most recent 10 messages.